[PyTorch Edge] Add test_lite_interpreter to fbsource xplat BUCK files

dhruvbird opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61997
* #61996
* #62306
* __->__ #62305

Currently, it's super time consuming to run a lite interpreter test from fbcode since it takes > 10 minutes to build. Recently, I haven't been able to do that either due to low disk space.

Having this test available in fbsource/xplat/ is a great win for productivity since I can re-run it in ~2 minutes even after significant changes!

I've had to disarm some tests that can only run in OSS of fbcode builds (since they need functionality that we don't include for on-device FB builds). They are disarmed using the macro `FB_XPLAT_BUILD`.

#accept2ship

Differential Revision: [D29954943](https://our.internmc.facebook.com/intern/diff/D29954943/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29954943/)!

dhruvbird committed:
[PyTorch Edge] Add test_lite_interpreter to fbsource xplat BUCK files

Currently, it's super time consuming to run a lite interpreter test from fbcode since it takes > 10 minutes to build. Recently, I haven't been able to do that either due to low disk space.

Having this test available in fbsource/xplat/ is a great win for productivity since I can re-run it in ~2 minutes even after significant changes!

I've had to disarm some tests that can only run in OSS of fbcode builds (since they need functionality that we don't include for on-device FB builds). They are disarmed using the macro `FB_XPLAT_BUILD`.

#accept2ship

Differential Revision: [D29954943](https://our.internmc.facebook.com/intern/diff/D29954943/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29954943/)!

[ghstack-poisoned]

cccclai commented on test/cpp/jit/test_lite_interpreter.cpp:
```diff
@@ -455,6 +455,7 @@ TEST(LiteInterpreterTest, BuiltinFunction) {
   AT_ASSERT(str == expected);
 }
 
+#if !defined FB_XPLAT_BUILD
```
It will be good to add some comments about why this flag is needed.

dhruvbird replied:
@cccclai Thanks! Updated with the commentary in the caffe2/BUCK file since it's FB specific.

cccclai reviewed (approved):
Thanks. Fix lint error and add some comments about the flag, then it's good to go.

dhruvbird committed:
Update on "[PyTorch Edge] Add test_lite_interpreter to fbsource xplat BUCK files"


Currently, it's super time consuming to run a lite interpreter test from fbcode since it takes > 10 minutes to build. Recently, I haven't been able to do that either due to low disk space.

Having this test available in fbsource/xplat/ is a great win for productivity since I can re-run it in ~2 minutes even after significant changes!

I've had to disarm some tests that can only run in OSS of fbcode builds (since they need functionality that we don't include for on-device FB builds). They are disarmed using the macro `FB_XPLAT_BUILD`.

#accept2ship

Differential Revision: [D29954943](https://our.internmc.facebook.com/intern/diff/D29954943/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29954943/)!

[ghstack-poisoned]

raziel commented on test/cpp/jit/test_lite_interpreter.cpp:
```diff
@@ -605,6 +614,7 @@ TEST(LiteInterpreterTest, GetByteCodeVersion) {
   auto version_v4 = _get_model_bytecode_version(test_model_file_v4);
   AT_ASSERT(version_v4 == 4);
 }
+#endif
```
```suggestion
#endif  // FB_XPLAT_BUILD
```

raziel commented on test/cpp/jit/test_lite_interpreter.cpp:
```diff
@@ -730,6 +740,7 @@ TEST(LiteInterpreterTest, BackPortByteCodeModelAllVersions) {
       expect_result_list,
       caffe2::serialize::kProducedBytecodeVersion);
 }
+#endif
```
```suggestion
#endif  // FB_XPLAT_BUILD 
```

raziel commented on test/cpp/jit/test_lite_interpreter.cpp:
```diff
@@ -596,6 +597,14 @@ TEST(LiteInterpreterTest, GetRuntimeByteCodeVersion) {
       caffe2::serialize::kMaxSupportedBytecodeVersion);
 }
 
+/**
```
If it's the same reason for all tests, maybe just move this comment to the top of the doc?

raziel commented on test/cpp/jit/test_lite_interpreter.cpp:
```diff
@@ -909,6 +921,7 @@ TEST(LiteInterpreterTest, DuplicatedClassTypeModuleInfo) {
   AT_ASSERT(module_debug_info_set.count("top(B).A0(A).aten::add"));
   AT_ASSERT(module_debug_info_set.count("top(B).A1(A).aten::add"));
 }
+#endif
```
```suggestion
#endif  // FB_XPLAT_BUILD
```

raziel commented on test/cpp/jit/test_lite_interpreter.cpp:
```diff
@@ -1285,7 +1299,9 @@ TEST(LiteInterpreterTest, DefaultArgsPinvSpecifyDefault) {
   inputs.push_back(input);
   testLiteModuleCompareResultTensors(m, inputs);
 }
+#endif
```
```suggestion
#endif  // FB_XPLAT_BUILD
```

raziel commented on test/cpp/jit/test_lite_interpreter.cpp:
```diff
@@ -1335,6 +1351,7 @@ Traceback of TorchScript (most recent call last):
   )";
   ASSERT_THROWS_WITH_MESSAGE(lite_m.forward(inputs), error_pattern);
 }
+#endif
```
```suggestion
#endif  // FB_XPLAT_BUILD
```

raziel reviewed (approved):
Thanks Dhruv.

Just some minor cleanups.

dhruvbird committed:
Update on "[PyTorch Edge] Add test_lite_interpreter to fbsource xplat BUCK files"


Currently, it's super time consuming to run a lite interpreter test from fbcode since it takes > 10 minutes to build. Recently, I haven't been able to do that either due to low disk space.

Having this test available in fbsource/xplat/ is a great win for productivity since I can re-run it in ~2 minutes even after significant changes!

I've had to disarm some tests that can only run in OSS of fbcode builds (since they need functionality that we don't include for on-device FB builds). They are disarmed using the macro `FB_XPLAT_BUILD`.

#accept2ship

Differential Revision: [D29954943](https://our.internmc.facebook.com/intern/diff/D29954943/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29954943/)!

[ghstack-poisoned]
